File: src/test/java/no/nav/fo/veilarbregistrering/registrering/bruker/OrdinaerBrukerRegistreringTestdataBuilder.java
package no.nav.fo.veilarbregistrering.registrering.bruker;

import no.nav.fo.veilarbregistrering.besvarelse.BesvarelseTestdataBuilder;
import no.nav.fo.veilarbregistrering.besvarelse.DinSituasjonSvar;
import no.nav.fo.veilarbregistrering.besvarelse.SisteStillingSvar;
import no.nav.fo.veilarbregistrering.besvarelse.StillingTestdataBuilder;

import java.time.LocalDateTime;

public class OrdinaerBrukerRegistreringTestdataBuilder {

    public static OrdinaerBrukerRegistrering gyldigBrukerRegistrering() {
        return gyldigBrukerRegistrering(LocalDateTime.now());
    }

    public static OrdinaerBrukerRegistrering gyldigBrukerRegistrering(LocalDateTime opprettetDato) {
        return new OrdinaerBrukerRegistrering()
                .setOpprettetDato(opprettetDato)
                .setSisteStilling(StillingTestdataBuilder.gyldigStilling())
                .setBesvarelse(BesvarelseTestdataBuilder.gyldigBesvarelse())
                .setTeksterForBesvarelse(TekstForSporsmalTestdataBuilder.gyldigeTeksterForBesvarelse());
    }

    public static OrdinaerBrukerRegistrering gyldigBrukerRegistreringUtenJobb() {
        return gyldigBrukerRegistrering().setSisteStilling(
                StillingTestdataBuilder.ingenYrkesbakgrunn()
        ).setBesvarelse(BesvarelseTestdataBuilder.gyldigBesvarelse()
                .setDinSituasjon(DinSituasjonSvar.ALDRI_HATT_JOBB)
                .setSisteStilling(SisteStillingSvar.INGEN_SVAR)
        );
    }


}

File: src/test/java/no/nav/fo/veilarbregistrering/registrering/bruker/OrdinaerBrukerRegistreringTestdataBuilder.kt
package no.nav.fo.veilarbregistrering.registrering.bruker

import no.nav.fo.veilarbregistrering.besvarelse.*
import no.nav.fo.veilarbregistrering.profilering.Profilering
import no.nav.fo.veilarbregistrering.profilering.ProfileringTestdataBuilder
import java.time.LocalDateTime

object OrdinaerBrukerRegistreringTestdataBuilder {

    @JvmStatic
    fun gyldigBrukerRegistrering(
        opprettetDato: LocalDateTime = LocalDateTime.now(),
        stilling: Stilling = StillingTestdataBuilder.gyldigStilling(),
        besvarelse: Besvarelse = BesvarelseTestdataBuilder.gyldigBesvarelse(),
        teksterForBesvarelse: List<TekstForSporsmal> = TekstForSporsmalTestdataBuilder.gyldigeTeksterForBesvarelse(),
        profilering: Profilering? = null,
    ): OrdinaerBrukerRegistrering {
        return OrdinaerBrukerRegistrering(
                opprettetDato = opprettetDato,
                sisteStilling = stilling,
                besvarelse = besvarelse,
                teksterForBesvarelse = teksterForBesvarelse,
                profilering = profilering,
        )
    }

    @JvmStatic
    fun gyldigBrukerRegistreringUtenJobb(opprettetDato: LocalDateTime = LocalDateTime.now()): OrdinaerBrukerRegistrering {
        return gyldigBrukerRegistrering(
            opprettetDato = opprettetDato,
            stilling = StillingTestdataBuilder.ingenYrkesbakgrunn(),
            besvarelse = BesvarelseTestdataBuilder.gyldigBesvarelse(
                dinSituasjon = DinSituasjonSvar.ALDRI_HATT_JOBB,
                sisteStilling = SisteStillingSvar.INGEN_SVAR,
            )
        )
    }
}